badgecheck/tasks/validation.py

```diff
@@ -152,30 +153,43 @@ def validate_primitive_property(state, task_meta):
     prop_name = task_meta.get('prop_name')
     prop_type = task_meta.get('prop_type')
     prop_value = node.get(prop_name)
-    required = bool(task_meta.get('prop_required'))
+    required = bool(task_meta.get('required'))
 
-    if not prop_value and required:
+    if prop_value is None and required:
```

Comment:
This allows for non-truthy prop values like `[]` and  `{}` which would be valid JSON values but still falsey. I take it that's what we want?